### ISSUE TITLE
Modular Backends

### ISSUE THREAD

**Issue by fdietze** [renesca/renesca#50]
provide a trait one can implement to use renesca.

for neo4j provide another trait for databases that support cypher.

stuff backends in their own libraries:

```
renesca-backend-cypher
renesca-backend-neo4j-REST
renesca-backend-neo4j-bolt
```

...

**Comment by PhilAndrew** [comment]
Yes would be good to support other graphs, for example DGraph https://github.com/dgraph-io/dgraph

**Comment by fdietze** [comment]
I didn't know about dgraph, thank you.

Because bolt is not async yet, it makes even more sense to have a modular backend. I'm reading about GraphQL right now...

**Comment by PhilAndrew** [comment]
Yes GraphQL is interesting ! I saw some Scala GraphQL lately on twitter.. 

GraphQL Neo4J https://github.com/neo4j-contrib/neo4j-graphql
Akka HTTP GraphQL https://github.com/sangria-graphql/sangria-subscriptions-example

**Comment by PhilAndrew** [comment]
Some more opinion, maybe what people want is a "Hibernate" for graphs. Hibernate was popular for a little while as a ORM for Java to most databases, then people decided that a ORM is an anti-pattern. Given that there are many graph databases, as a Scala programmer I would like to deal with one API no matter what the database is. Does GraphQL fit into that. Best would be that the one API can work on graphs all over the place, in RAM, in ScalaJS, in Neo4j.

About Neo4j-OGM https://github.com/neo4j/neo4j-ogm - I feel you would be better to integrate and complement it rather than compete with.
